src/v/coproc/script_context.cc

```diff
@@ -168,6 +168,13 @@ ss::future<> script_context::send_request(
 }
 
 bool script_context::is_up_to_date() const {
+    if (_routes.empty()) {
+        /// Since this method is only used for unit tests, its never desired to
+        /// return true in the case the script is technically "up to date"
+        /// because it contains no work. It is likely due to recieve data to
+        /// process shortly.
+        return false;
```

Comment:
It is a little bit confused for me. Looks like broken logic?
If it is ok, LGTM

Reply:
The comment is explaining that this is all due to how one defines "completion". If there is nothing registered, one could say its finished because there's no work, or one could say it's not finished because it is expecting some work and nothing has yet been registered. 

I am choosing the definition of finished to be false when there's no work in order for this to easily make a failing test pass, and since its only used for testing and the definition could be defined either way, i think its an acceptable solution.